src/plugins/data_source/server/client/client_pool.ts

```diff
@@ -21,23 +22,28 @@ export interface OpenSearchClientPoolSetup {
  * It reuse TPC connections for each OpenSearch endpoint.
```

Comment:
with sigv4, it's reusing connection only when `datasource` hasn't been updated. (cache key for aws client is `endpoint + lastUpdatedTime`)

Reply:
can you help me to recall why we need `lastUpdateTime` in the cache key?

Reply:
Opensearch client doesn't allow spawning child with connection class(in sigv4 use case, the SigV4 signer). So we have to create a new aws opensearch client for a datasource, instead of spawning a child from a root client. If we only use `endpoint` as cache key, cache hit could happen for an **updated** data source, outdated client will be fetched.  By using `endpoint + lastUpdatedTime of savedObject` for `awsClientCache`, we can make sure it will be a cache miss, and by logic it creates new client. Also it re-uses client for non-changed sigv4 datasources.